include/pybind11/numpy.h

```diff
@@ -165,16 +165,16 @@ class dtype : public object {
 public:
     PYBIND11_OBJECT_DEFAULT(dtype, object, detail::npy_api::get().PyArrayDescr_Check_);
 
-    dtype(const buffer_info &info) {
+    explicit dtype(const buffer_info &info) {
         dtype descr(_dtype_from_pep3118()(PYBIND11_STR_TYPE(info.format)));
         m_ptr = descr.strip_padding().release().ptr();
     }
 
-    dtype(std::string format) {
+    explicit dtype(std::string format) {
         m_ptr = from_args(pybind11::str(format)).release().ptr();
     }
 
-    dtype(const char *format) : dtype(std::string(format)) { }
+    explicit dtype(const char *format) : dtype(std::string(format)) { }
```

Comment:
A bit later to the party, but this one was actually kind of meant to be implicit, so you could have numpy-like constructs and use a string literal every time a dtype is required (and it's only required in `array` methods/ctors).E.g.:

```cpp
py::array(100, "|S4");
py::array({2, 3}, "i");
```